kcidb/__init__.py

```diff
@@ -253,8 +253,13 @@ def validate_main():
     parser = misc.ArgumentParser(description=description)
     misc.argparse_schema_add_args(parser, "validate against")
     args = parser.parse_args()
-    for data in misc.json_load_stream_fd(sys.stdin.fileno()):
-        print(args.schema_version.validate(data))
+    misc.json_dump_stream(
+        (
+            args.schema_version.upgrade(io.schema.validate(data), copy=False)
```

Comment:
This changes the function and the `kcidb-validate` tool to also upgrade the data, which is not in the issue you're trying to fix: https://github.com/kernelci/kcidb/issues/145

Your first attempt was basically fine, and I'm just asking you to make the tool's interface match the other tools.

Ping me on Slack, if you need help figuring this out.